Code/app/src/main/java/com/cmpt362/fitcheck/ui/calendar/EditActivity.kt

```diff
@@ -0,0 +1,164 @@
+        tagEditText.setAdapter(adapter)
+        tagEditText.setOnKeyListener(View.OnKeyListener { v, keyCode, event ->
+            if ((keyCode == KeyEvent.KEYCODE_ENTER) && (event.action == KeyEvent.ACTION_UP)) {
+                val value = tagEditText.text.toString()
+                addChipToGroup(value)
+                tagArray.add(value)
+                val inputMethodManager = getSystemService(AppCompatActivity.INPUT_METHOD_SERVICE) as InputMethodManager
+                inputMethodManager.hideSoftInputFromWindow(v.windowToken, 0)
+                tagEditText.text.clear()
+                return@OnKeyListener true
+            }
+            return@OnKeyListener false
+        })
+        Firebase.getPhoto(year, month, day, imageView, notesText, chipGroup, this)
+    }
+
+    private fun addChipToGroup(tag: String) {
+        val chip = Chip(this)
+        chip.text = tag
+        chip.chipIcon = ContextCompat.getDrawable(this, R.drawable.ic_launcher_background)
+        chip.isChipIconVisible = false
+        chip.isCloseIconVisible = true
+        chip.isClickable = false
+        chip.isCheckable = false
+        chipGroup.addView(chip as View)
+        chip.setOnCloseIconClickListener {
+            chipGroup.removeView(chip as View)
+            tagArray.remove(tag)
+        }
+    }
+
+    fun convertToString(month: Int): String {
+        var monthString = "default"
+        when (month) {
+            1 -> monthString = "Jan"
+            2 -> monthString = "Feb"
+            3 -> monthString = "Mar"
+            4 -> monthString = "Apr"
+            5 -> monthString = "May"
+            6 -> monthString = "June"
+            7 -> monthString = "July"
+            8 -> monthString = "Aug"
+            9 -> monthString = "Sept"
+            10 -> monthString = "Oct"
+            11 -> monthString = "Nov"
+            12 -> monthString = "Dec"
+        }
+        return monthString
+    }
+
+    /**
+     * When save button is click, save the upload and finish the activity
+     */
+    fun onSaveUpload(view: View){
+        val tagReference = Firebase.getTag()
+        val eventListener = object : ValueEventListener {
+            override fun onDataChange(dataSnapshot: DataSnapshot) {
+                for (ds: DataSnapshot in dataSnapshot.children) {
+                    val keyValue = ds.getValue(String::class.java)
+                    databaseTagArray.add(keyValue!!)
+                }
+            }
+            override fun onCancelled(databaseError: DatabaseError) {}
+        }
+        tagReference.child("tags").addListenerForSingleValueEvent(eventListener)
+
+        for(item in tagArray){
+            // if the tag DOES NOT exist in the database then add it
+            if(databaseTagArray.indexOf(item) == -1) {
+                val newData = tagReference.child("tags").push()
+                newData.setValue(item)
+            }
+        }
```

Comment:
I don't know how but the tagArray only contains the new tags that the user input. It doesn't contain the previously saved tags so I think that's why they get deleted after editing.

Let me know if this doesn't make sense I can try to explain better.